purchases/src/main/kotlin/com/revenuecat/purchases/UpgradeInfo.kt

```diff
@@ -1,15 +1,14 @@
 package com.revenuecat.purchases
 
-import com.android.billingclient.api.BillingFlowParams
+import com.revenuecat.purchases.models.GoogleProrationMode
 
 /**
  * This object holds the information used when upgrading from another sku.
  * @property oldProductId The old product ID to upgrade from.
- * @property prorationMode The [BillingFlowParams.ProrationMode] to use when upgrading the given oldSku. Defaults to
- * [BillingFlowParams.ProrationMode.IMMEDIATE_WITHOUT_PRORATION].
+ * @property googleProrationMode The [GoogleProrationMode] to use when upgrading the given oldProductId. Defaults to
+ * [GoogleProrationMode.IMMEDIATE_WITHOUT_PRORATION].
  */
 data class UpgradeInfo @JvmOverloads constructor(
     val oldProductId: String,
-    @BillingFlowParams.ProrationMode val prorationMode: Int =
-        BillingFlowParams.ProrationMode.IMMEDIATE_WITHOUT_PRORATION
+    val googleProrationMode: GoogleProrationMode = GoogleProrationMode.IMMEDIATE_WITHOUT_PRORATION
```

Comment:
Hmm I wonder if we should make this nullable and just not set a default, so we use Google's default instead... I think this is ok though.

Reply:
Google's default is not right / also not clear 😬 The two default values that are chosen on Google Play Console (which are `IMMEDIATE_WITH_TIME_PRORATION` and `DEFERRED`) don't fully work for us.
- `IMMEDIATE_WITH_TIME_PRORATION` seems to not work when switching between base plans
- `DEFERRED` isn't fully supported with the API right now

Billing library also says that `IMMEDIATE_WITH_TIME_PRORATION` is the default but it _should_ probably be using the default that are set in Google Play Console... but it's not 🙈 

`IMMEDIATE_WITHOUT_PRORATION` has been tested to work 100% of the time so this is what we need to default to right now 🤷‍♂️

Reply:
Ahhh interesting... Makes sense then. Thanks for the explanation!